src/store.ts

```diff
@@ -0,0 +1,107 @@
+// SPDX-FileCopyrightText: 2022 Carl Schwan <carl@carlschwan.eu>
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+import { defineStore } from 'pinia'
+import axios from '@nextcloud/axios'
+import { generateUrl } from '@nextcloud/router'
+import { computed, ref } from 'vue'
+import type { Ref } from 'vue'
+
+interface StackElement {
+
+}
+
+interface Profile {
+
+}
+
+interface Profiles {
+	[index: string]: Profile;
+}
+
+/**
+ * Read a file to string
+ * @param file
+ */
+function readFile(file: Blob): Promise<string> {
+	return new Promise((resolve, reject): void => {
+		const reader: FileReader = new FileReader()
+		reader.readAsText(file, 'UTF-8')
+		reader.onload = function(): void {
+			if (reader.result) {
+				resolve(reader.result.toString())
+			} else {
+				reject(new Error('File output is null'))
+			}
+		}
+		reader.onerror = function(): void {
+			reject(new Error('error reading file'))
+		}
+	})
+}
+
+export const useStore = defineStore('main', () => {
+	const profiles: Ref<Profiles> = ref({})
```

Comment:
IMHO you do not need to explicitly type here and just use the inferred type like

```suggestion
	const profiles = ref<Profiles>({})
```

But both work :shrug: